shanalikhan/code-settings-sync
# Add support for disabled extensions

#### Short description of what this resolves:

Takes account disabled extensions when uploading or downloading the gist file.

#### Changes proposed in this pull request:

- `CreateExtensionList` now uses `code --list-extensions` to get enabled and disabled extensions;
- When `download` executes it verifies the extension object's enabled property . If false calls `code --disable-extension` with the extension name.

**Fixes**: #143 

#### How Has This Been Tested?




Tested on a Macbook with VSCode.

- Open `code-settings-sync` project;
- Disable an extension; 
- While debugging (`F5`) inside this branch execute `Upload / Update settings`;
- Then check in your gist if the current extension is disabled;
- Leave the debugging mode and uninstall the same extension;
- Open the debugging mode and execute `Download settings`;
- Check if it installs the extension and disables it automatically.

#### Checklist:
